flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1232,21 +1232,31 @@ def run_to_convergence(self, desired_datasets, max_iterations=4):
 
     @given(
```

Comment:
In an ideal world the fact the next dataset is related to the initial one would be visible to Hypothesis, as it was before... This change reduces the search space as a result, since we get more duplicate iterations; limited number of cases we have in each run are wasted on generating effectively duplicate random variants of next dataset.

Possibly https://hypothesis.readthedocs.org/en/latest/data.html#chaining-strategies-together might help? Not sure.